modules/Sources/Features/SendFlow/SendFlowStore.swift

```diff
@@ -199,63 +212,94 @@ public struct SendFlowReducer: Reducer {
                 state.destination = nil
                 state.isSending = false
                 return .none
+
+            case .partialProposalError:
+                return .none
                 
+            case let .proposal(proposal):
+                state.proposal = proposal
+                return .none
+        
             case let .updateDestination(destination):
                 state.destination = destination
                 return .none
 
             case .reviewPressed:
-                state.destination = .sendConfirmation
-                return .none
+                return .send(.updateDestination(.sendConfirmation))
+                
+                return .run { [state] send in
+                    do {
+                        let recipient = try Recipient(state.address, network: zcashSDKEnvironment.network.networkType)
+                        
+                        let memo: Memo?
+                        if state.transactionAddressInputState.isValidTransparentAddress {
+                            memo = nil
+                        } else if let memoText = state.addMemoState ? state.memoState.text : nil {
+                            memo = memoText.data.isEmpty ? nil : try Memo(string: memoText.data)
+                        } else {
+                            memo = nil
+                        }
+
+                        let proposal = try await sdkSynchronizer.proposeTransfer(0, recipient, state.amount, memo)
+                        
+                        await send(.proposal(proposal))
+                        await send(.updateDestination(.sendConfirmation))
+                    } catch {
+                        await send(.sendFailed(error.toZcashError()))
+                    }
+                }
 
             case .sendPressed:
+                return .send(.sendPartial(["ba5113d9e78c885bdb03d44f784dc35aa0822c0f212f364b5ffc994a3e219d1f", "asdfasfqerqrafdb03d44f784dc35aa0822c0f212f364b5ffc99werwrt5wewerqwtr"]))
+                state.isSending = true
+
+                guard let proposal = state.proposal else {
+                    return .send(.sendFailed("missing proposal".toZcashError()))
```

Comment:
In theory - never but I'm still handling this scenario just in case. 

The thing is I need to pass a Proposal in the TCA (composable architecture) and the initial state is nil (Proposal?). Once the proposal is available, I pass it in TCA actions but the state.proposal is still an Optional type. I need to unwrap it before it can be used in send action and that is exactly what's going on here. Guard is unwrapping it but if that fails or proposal is nil, something must happen other than `return .none` side effect. That would mean UX when nothing happens when a user taps on send button.